api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingBatchUpdateService.java

```diff
@@ -0,0 +1,75 @@
+package org.pmiops.workbench.billing;
+
+import com.google.common.collect.Iterables;
+import java.util.HashSet;
+import java.util.List;
+import java.util.logging.Logger;
+import javax.inject.Provider;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.db.dao.UserDao;
+import org.pmiops.workbench.db.model.DbUser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.util.StreamUtils;
+import org.springframework.stereotype.Service;
+
+/**
+ * Class to call the {@link FreeTierBillingService} with batches of users. This ensures that
+ * FreeTierBillingService will commit the transaction for a smaller batch of users instead of
+ * processing all users in one transaction and eventually timing out. See RW-6280
+ */
+@Service
+public class FreeTierBillingBatchUpdateService {
+
+  private static final Logger logger =
+      Logger.getLogger(FreeTierBillingBatchUpdateService.class.getName());
+
+  private final UserDao userDao;
+  private final FreeTierBillingService freeTierBillingService;
+  private final Provider<WorkbenchConfig> workbenchConfigProvider;
+
+  private static final int MIN_USERS_BATCH = 5;
+  private static final int MAX_USERS_BATCH = 999;
```

Comment:
We're limited to 999 because we use it in an IN clause in sql. Please check WorkspaceDao.getWorkspaceCostViews method. I think the min size can be relaxed to be 1, although I had good performance with batches of 5 in test env. 
I think we can keep them as is for now

I actually changed the implementation of the method getFreeTierCronUserBatchSize to be more clear and I'll push my changed shortly.